### PR TITLE
Parameterize

### LINKED ISSUE #13: Add new parameters
Add new config parameters:
- [x] testnet, simnet (default: mainnet)
- [x] lnd dir
- [x] certs path
- [x] macaroon path
- [x] dcrlnfaucet config path

### PR DESCRIPTION
**Pull request by fguisso**
Add new parameters:
- testnet, simnet (default: mainnet)
- certs path
- macaroon path
- dcrlnfaucet config path
- disable generate invoice form
- disable pay invoice form

resolve #13

### CODE REVIEW

File: faucet.go [hunk 1]
```diff
@@ -198,24 +199,51 @@ type lightningFaucet struct {
...
 
 // newLightningFaucet creates a new channel faucet that's bound to a cluster of
 // lnd nodes, and uses the passed templates to render the web page.
-func newLightningFaucet(lndNode string,
+func newLightningFaucet(cfg *config,
 	templates *template.Template) (*lightningFaucet, error) {
 
 	// First attempt to establish a connection to lnd's RPC sever.
-	creds, err := credentials.NewClientTLSFromFile(tlsCertPath, "")
+	creds, err := credentials.NewClientTLSFromFile(cfg.TLSCertPath, "")
```
Comment (matheusd) [inline faucet.go]: You should use `cleanAndExpandPath` here as well.

File: faucet.go [hunk 2]
```diff
@@ -801,26 +839,17 @@ func (l *lightningFaucet) CloseAllChannels() error {
...
-
 // generateInvoice is a hybrid http.Handler that handles: the validation of the
 // generate invoice form, rendering errors to the form, and finally generating
 // invoice if all the parameters check out.
 func (l *lightningFaucet) generateInvoice(homeTemplate *template.Template,
 	homeState *homePageContext, w http.ResponseWriter, r *http.Request) {
+
+	// Disable generate invoice if user set this parameter
+	if l.disableGenerateInvoices {
+		return
```
Comment (matheusd) [inline faucet.go]: This (and the corresponding call in payInvoice) should return an http 403 error.

File: config.go [hunk 3]
```diff
@@ -118,8 +150,36 @@ func loadConfig() (*config, []string, error) {
...
+		str := "%s: mainnet, testnet and simnet params can't be " +
+			"used together -- choose one of the three"
+		err := fmt.Errorf(str, funcName)
+		fmt.Fprintln(os.Stderr, err)
+		fmt.Fprintln(os.Stderr, usageMessage)
+		return nil, nil, err
+	}
+
+	// Update LndNode using the activeNetParams
+	cfg.LndNode = "localhost:" + activeNetParams.rpcPort
```
Comment (matheusd) [inline config.go]: This is overriding `cfg.LndNode`, so if I run with `--lnd_node 10.20.30.40:10000` the faucet won't be able to connect to the lnd node.

**Comment by fguisso** [comment]
fix it

File: config.go [hunk 4]
```diff
@@ -174,8 +173,11 @@ func loadConfig() (*config, []string, error) {
 		return nil, nil, err
 	}
 
-	// Update LndNode using the activeNetParams
-	cfg.LndNode = "localhost:" + activeNetParams.rpcPort
+	// If user don't sets the LndNode, then update LndNode
+	// using the activeNetParams to deafult (localhost:<default_port>).
```
Comment (matheusd) [inline config.go]: "If user didn't set the LndNode to connect to then use the default port for the active network (localhost:<default_port>)."

**Comment by fguisso** [comment]
I forgot to update sample.conf with new parameters, one minute please! 😮

**Pull request merged**